### PR TITLE
Function to get the image from vcenter added

### PR DESCRIPTION
**Pull request by avnish30jn**
**Jira ID**

C3-64

**Problem**

No functionality to get the templates in vsphere cloud

**Solution**

Added support for returning the templates as the image list

**Testing**

Unit testing done - Returns the templates and datacenter in the vcenter server

**O/P**

{"Apporbit-Lab":["centos73-secured","rhel72-template","centos73-update2","centos73-update3","centos72-250GB-secure-ready","redhat72-template","jenkin-registry-template","apporbit-apps-template","windows-2012-template","centos72-template-2","centos72-20GB-ready","Oracle-linux67-template","local-apporbit-visor-temp-datastore1","windows-2012-r2-template","-datastore1","repos-template-Feb26","private_network_template-datastore1","template-2.1-datastore1","apporbit-2.0-datastore1"]}

**Order for Reviewing**
below PR needs to be merged first
https://github.com/apporbit/libretto/pull/10

### CODE REVIEW

**Comment by sjpotter** [comment]
looks clean to me, though as I commented on the othe PR, wondering if we should be using the vm struct like this

File: virtualmachine/vsphere/vm.go [hunk 1]
```diff
@@ -779,3 +780,47 @@ func DeleteTemplate(vm *VM) error {
...
+			vmsMor = append(vmsMor, vm.Reference())
+		}
+		// get the vm names and config
+		var allVmsMo []mo.VirtualMachine
+		err = vm.collector.Retrieve(vm.ctx, vmsMor, []string{"name", "config"}, &allVmsMo)
+		if err != nil {
+			return nil, err
+		}
+		// generate response for the images in datacenter. In the response map
+		// the key is the datacenter name and value is the list ofim images in datacenter
```
Comment (pankajtandon12) [inline virtualmachine/vsphere/vm.go]: ofim ->of

**Review by pankajtandon12 — APPROVED** [review]
LGTM.
Consumer need to take care of adding a filter for visor vm-template nomenclature. Raise jira ticket for same.

File: virtualmachine/vsphere/vm.go (hunk 1 above)
Comment (avnish30jn) [reply virtualmachine/vsphere/vm.go]: Corrected

**Pull request merged**